ecs-init/engine/engine.go

```diff
@@ -180,15 +183,23 @@ func (e *Engine) StartSupervised() error {
 			return engineError("could not start Agent", err)
 		}
 		log.Infof("Agent exited with code %d", agentExitCode)
-		if agentExitCode == upgradeAgentExitCode {
+
+		switch agentExitCode {
+		case upgradeAgentExitCode:
 			err = e.upgradeAgent()
 			if err != nil {
 				log.Error("could not upgrade agent", err)
+			} else {
+				continue
 			}
+		case terminalFailureAgentExitCode:
+			return errors.New("agent exited with terminal exit code")
+		case terminalSuccessAgentExitCode:
+			return nil
 		}
-	}
-	if agentExitCode == terminalFailureAgentExitCode {
-		return errors.New("agent exited with terminal exit code")
+		d := backoff0.Duration()
+		log.Infof("Service failed to start, sleeping %s and retrying", d)
```

Comment:
This shouldn't be an informational log line, its a warning that `ecs-init` wasn't able to start the agent but is going to retry again. Maybe use `log.Warnf` and a message like `Agent container startup failed, retrying startup in %s`.

Reply:
sure, I'll change it to a warning 👍